Docs/lambda-annotations-design.md

```diff
@@ -0,0 +1,358 @@
+    * A piece of code that runs before or after any request in the project
+* Integrated dependency injection
+    * A central location for registering all of the dependencies for the request in the project.
+* Authentication
+    * Define policies and add annotations to request paths that ensure the caller has the permissions for the policies.
+
+
+Here is an example of a ASP.NET Core REST controller doing similar logic to the low level example mention before.
+
+
+```
+[Route("[controller]")]
+public class MathController : ControllerBase
+{
+    [HttpGet("/plus/{x}/{y}")]
+    public int Plus(int x, int y)
+    {
+        return x + y;
+    }
+
+    [HttpGet("/subtract/{x}/{y}")]
+    public int Subtract(int x, int y)
+    {
+        return x - y;
+    }
+}
+```
+
+
+There are a few significant negative side effect for writing REST APIs for Lambda using ASP.NET Core. 
+
+* Significant impact to cold start due to size of the framework and reliance on reflection.
+* The common features of ASP.NET Core for REST APIs are supported in Lambda but a significant number of the features are not and there is no clear indication what is and isn�t supported.
+* The usage of API Gateway is dumbed down because the API Gateway REST API is defined as a single wild card resource path letting ASP.NET Core handling the routing. This takes away a lot of the features of API Gateway like per resource path IAM permissions and memory sizing. 
+* ASP.NET Core can only be used for defining REST APIs. It can not be used for other types of Lambda functions like S3 events.
+
+## Goals for a better experience
+
+To improve the experience of writing Lambda functions we need an experience that is similar to what developers are used to with ASP.NET Core without the negative side effects. The major goals are:
+
+* An experience similar to what .NET developers are used in frameworks like ASP.NET Core
+    * Although it will follow similar patterns we will not reuse the same classes as ASP.NET Core to avoid an implicit contract agreements between the developer and the framework.
+* No significant impact to cold start
+    * Measuring with a prototype the plus math operation cold start was in the **~320ms** the same as the basic Lambda programming model. Using ASP.NET Core the cold start was **~1,200ms**.
+* Can be used for more scenarios then just REST APIs
+* Underlying power of AWS services is still intuitively available
+* Keeping code and CloudFormation template in sync
+
+
+
+## The new experience for writing .NET Lambda functions
+
+To create a new experience for developing .NET Lambda functions we will use a combination of .NET attributes developer apply to their code and C# source generators to generate code and CloudFormation snippets at compile time. This will remove the need for reflection at runtime and the only runtime overhead component will be the inclusion of the new .NET attributes and interfaces. The .NET attributes will have negligible effect to cold start.
```

Comment:
```suggestion
To create a new experience for developing .NET Lambda functions we will use a combination of .NET attributes developer applies to their code and C# source generators to generate code and CloudFormation snippets at compile time. This will remove the need for reflection at runtime and the only runtime overhead component will be the inclusion of the new .NET attributes and interfaces. The .NET attributes will have negligible effect to cold start.
```

Reply:
Fixed but slightly different by pluralizing `developer`.